Explorer/Assets/DCL/Backpack/BackpackBus/BackpackBusController.cs

```diff
@@ -89,17 +100,42 @@ private void HandleFilterCommand(BackpackFilterCommand command) =>
         private void HandleSelectWearableCommand(BackpackSelectWearableCommand command)
         {
             if (wearableStorage.TryGetElement(command.Id, out IWearable wearable))
-                backpackEventBus.SendWearableSelect(wearable);
+                SelectWearable(wearable);
+            else
+                FetchWearableByPointerAndExecuteAsync(command.Id, SelectWearable).Forget();
         }
 
-        private void HandleEquipWearableCommand(BackpackEquipWearableCommand command)
+        private void SelectWearable(IWearable wearable) =>
+            backpackEventBus.SendWearableSelect(wearable);
+
+        private async UniTaskVoid FetchWearableByPointerAndExecuteAsync(string pointer, Action<IWearable> onWearableFetched)
         {
-            if (!wearableStorage.TryGetElement(command.Id, out IWearable wearable))
+            try
             {
-                ReportHub.LogError(new ReportData(ReportCategory.WEARABLE), $"Cannot equip wearable, not found: {command.Id}");
-                return;
+                urnRequest[0] = new URN(pointer);
+
+                await UniTask.WhenAll(wearablesProvider.RequestPointersAsync(urnRequest, BodyShape.MALE, fetchWearableCts.Token),
+                        wearablesProvider.RequestPointersAsync(urnRequest, BodyShape.FEMALE, fetchWearableCts.Token));
+
+                if (wearableStorage.TryGetElement(pointer, out IWearable wearable))
```

Comment:
You don't need to retrieve from cache as you can get the result from `UniTask` directly